contrib/tracking.go

```diff
@@ -0,0 +1,256 @@
+package contrib
+
+/*
+#include "tracking.h"
+*/
+import "C"
+import (
+	"image"
+
+	"gocv.io/x/gocv"
+)
+
+// Tracker - the base interface for object tracking
+//
+// s.a: https://docs.opencv.org/master/d0/d0a/classcv_1_1Tracker.html
+type Tracker interface {
+	// Init initializes the tracker with a known bounding box that surrounded the target
+	// note: this can only be called once. if you loose the object, you have to Close() the instance,
+	// create a new one, and call Init on it again
+	//
+	// https://docs.opencv.org/master/d0/d0a/classcv_1_1Tracker.html#a4d285747589b1bdd16d2e4f00c3255dc
+	Init(image gocv.Mat, boundingBox image.Rectangle) bool
+
+	// Update the tracker, return new bounding box and a bool, if it lost it.
+	//
+	// https://docs.opencv.org/master/d0/d0a/classcv_1_1Tracker.html#a549159bd0553e6a8de356f3866df1f18
+	Update(image gocv.Mat) (image.Rectangle, bool)
+}
+
+//
+// available implementations in gocv are:
+//
+// TrackerMil;
+// TrackerBoosting;
+// TrackerMedianFlow;
+// TrackerTld;
+// TrackerKcf;
+// TrackerMosse;
+// TrackerCsrt;
+//
+
+// (private) implementation helpers for init and update (saves some boilerplate code)
+func tracker_Init(trk C.Tracker, img gocv.Mat, boundingBox image.Rectangle) bool {
+	cBox := C.struct_Rect{
+		x:      C.int(boundingBox.Min.X),
+		y:      C.int(boundingBox.Min.Y),
+		width:  C.int(boundingBox.Size().X),
```

Comment:
How different `boundingBox.Size().X` and `boundingbox.Max.X` are?

Reply:
Max.X = Min.X + width

(opencv's rects are [x,y,w,h] go's are [x0,y0, x1,y1])

Reply:
oh ok, thanks for the explanation